### ISSUE TITLE
Multiplicities appear in valid hole fit messages

### ISSUE THREAD

**Issue by aspiwack** [tweag/ghc#217]
What seems to be happening here, is that in the `with foo` bit of the valid hole fit printing, `foo` printed out from Core, with type applications visibles. All the type arguments are printed as `@…` which is, I think, incorrect for inferred type arguments (it's not reparsable for starters).

In our case, it makes multiplicities appear when `foo` is a constructor because of the constructor-are-multiplicity-polymorphic-thing. If this is not considered an error upstream, then we will need a hack to hide multiplicities as per the terms of the proposal acceptance.

This makes all of

```
make TEST="T14273 T14343 T14343b type_in_type_hole_fits valid_hole_fits valid_hole_fits_interactions" test
```

to have multiplicities in their output.

May be related to [D5314](https://phabricator.haskell.org/D5314).

**Comment by aspiwack** [comment]
The relevant code is, I think, in: `TcHoleErrors.pprHoleFit`.

**Comment by monoidal** [comment]
Reported upstream at https://gitlab.haskell.org/ghc/ghc/issues/16456.

**Comment by monoidal** [comment]
Fixed in #365.
